project-base/storefront/helpers/pagination/loadMore.ts

```diff
@@ -61,3 +66,55 @@ export const getPreviouslyQueriedProductsFromCache = (
 
     return cachedPartOfProducts;
 };
+
+export const getOffsetPageAndLoadMoreBasedOnLoadedProductsLimit = (
+    currentPage: number,
+    currentLoadMore: number,
+    pageSize = DEFAULT_PAGE_SIZE,
+) => {
+    const loadedProductsDifference = (currentLoadMore + 1) * pageSize - PRODUCT_LIST_LIMIT;
+
+    if (loadedProductsDifference <= 0) {
+        return undefined;
+    }
+
+    const pageOffset = Math.ceil(loadedProductsDifference / pageSize);
+
+    return {
+        updatedPage: currentPage + pageOffset,
+        updatedLoadMore: currentLoadMore - pageOffset,
+    };
+};
+
+export const getRedirectWithOffsetPageIfTooManyLoadedProducts = (
```

Comment:
Cannot we shorten the names of those two functions? Those long names are counterproductive in my opinion. Since until I read function name I would rather simply open this function and maybe faster read code itself to understand what it does. Maybe `getRedirectWithOffsetPage` would do the job as well? And `getRedirectWithOffsetPage`? Since I don't have to know about how exactly is this offset or redirect calculated.

Reply:
Renamed it to getOffsetPageAndLoadMore and getRedirectWithOffsetPage.